pkg/build/builder/daemonless.go

```diff
@@ -311,118 +328,178 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 		PullPushRetryDelay:      DefaultPushOrPullRetryDelay,
 	}
 
-	_, _, err := imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
+	_, _, err = imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
 	return err
 }
 
-func generateTransientMounts() []string {
-	mounts := []string{}
-	mounts = appendRHSMMount(defaultMountStart, mounts)
-	mounts = appendETCPKIMount(defaultMountStart, mounts)
-	mounts = appendRHRepoMount(defaultMountStart, mounts)
-	mounts = appendCATrustMount(mounts)
+type TransientMounts map[string]string
+
+func (t TransientMounts) append(mount string) error {
```

Comment:
Why not accept an object that mirrors or extends the `BuildVolumeMount` object in the api? That way `append` can encode the data into the internal map, with the appropriate options added on.

```suggestion
type TransientMount struct {
   buildv1.BuildVolumeMount
   // additional buildah mount options not exposed to the Build API
  NoDev bool 
  NoExec bool
  NoSuid bool
}

func (t TransientMounts) append(mount TransientMount) error {
```